x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/SqlTranslatorHandler.java

```diff
@@ -34,12 +35,12 @@ public Query asQuery(Expression e) {
     }
 
     @Override
-    public Query wrapFunctionQuery(ScalarFunction sf, Expression field, Query q) {
-        if (field instanceof StDistance && q instanceof GeoDistanceQuery) {
-            return ExpressionTranslator.wrapIfNested(q, ((StDistance) field).left());
+    public Query wrapFunctionQuery(ScalarFunction sf, Expression field, Supplier<Query> querySupplier) {
+        if (field instanceof StDistance && querySupplier instanceof GeoDistanceQuery) {
```

Comment:
Shouldn't it be: `querySupplier.get()`?

Reply:
Thanks, good catch! Seems there is no test, good time to add one.